File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/04 Print N to 1/print_nto1.py
# https://www.geeksforgeeks.org/problems/print-n-to-1-without-loop/1

class Solution:
    
    def print_nto1(self, n):
        # Base condition
        if n == 0:
            return
        print(n, end=" ")
        self.print_nto1(n - 1)

# Example usage
solution = Solution()
solution.print_nto1(5)  # Prints: 1 2 3 4 5 

File: Step 1 - Learn the Basics/1.5 Learn Basic Recursion/04 Print N to 1/print_nto1.java
// https://www.geeksforgeeks.org/problems/print-n-to-1-without-loop/1

// User function Template for Java

class Solution {

    public void print_nto1(int n) {
        // Your code here
        if(n==0)
            return;
        System.out.print(n+" ");
        print_nto1(n-1);
        
    }
}
